modules/diskq/diskq-global-metrics.c

```diff
@@ -197,7 +197,63 @@ diskq_global_metrics_watch_dir(const gchar *dir)
   g_mutex_lock(&self->lock);
   {
     if (!g_hash_table_contains(self->dirs, dir))
-      g_hash_table_insert(self->dirs, g_strdup(dir), NULL);
+      {
+        GHashTable *acquired_files = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, NULL);
+        g_hash_table_insert(self->dirs, g_strdup(dir), acquired_files);
+      }
   }
   g_mutex_unlock(&self->lock);
 }
+
+void
+diskq_global_metrics_file_acquired(const gchar *abs_filename)
+{
+  DiskQGlobalMetrics *self = &diskq_global_metrics;
+
+  gchar *dir = g_path_get_dirname(abs_filename);
+  gchar *filename = g_path_get_basename(abs_filename);
+
+  g_mutex_lock(&self->lock);
+  {
+    GHashTable *acquired_files = g_hash_table_lookup(self->dirs, dir);
+    if (!acquired_files)
+      {
+        g_mutex_unlock(&self->lock);
+        g_free(filename);
+        goto exit;
```

Comment:
If you don't want to force-watch, a `g_assert_not_reached()` might save us some future trouble.

Reply:
I started with that and the unit tests crashed on that, because they don't call watch, but call acquire_queue. I can add a watch to the unit tests, too, but it would seem strange IMO.

Reply:
From a clean-code perspective, if the test does not call watch() but only calls acquire(), that's an invalid usage of the API and the test should probably catch that.

it's either we do a force-watch, or we don't allow calling this method without a previous watch() call. With the current code, calling `diskq_global_metrics_file_acquired()` without a previous watch() call will misfunction.

Reply:
I agree, fixed.